### PR TITLE
[decrypt_all] Check if file needs to decrypted or not for speed up large oc setups.

### PR DESCRIPTION
**Pull request by PVince81**
Salvaged commits from https://github.com/owncloud/core/pull/22955 by @chriseqipe

@schiesbn @icewind1991 can you check whether this is the correct approach for checking the encrypted flag ?

### CODE REVIEW

File: lib/private/Encryption/DecryptAll.php [hunk 1]
```diff
@@ -213,9 +214,14 @@ protected function decryptUsersFiles($uid, ProgressBar $progress, $userCount) {
 					try {
 						$progress->setMessage("decrypt files for user $userCount: $path");
 						$progress->advance();
-						if ($this->decryptFile($path) === false) {
+						if ((int)$file->getData()['encrypted'] === 0) {
```
Comment (schiesbn) [inline lib/private/Encryption/DecryptAll.php]: use `if ($file->isEncrypted()) {` instead

**Comment by PVince81** [comment]
@chriseqipe are you interested in finishing this fix ? If yes, please see the review comments above.
If not, let us know so someone can take over.

Thanks!

File: lib/private/Encryption/DecryptAll.php (hunk 1 above)
Comment (PVince81) [inline lib/private/Encryption/DecryptAll.php]: done

**Comment by PVince81** [comment]
Adjusted the PR.

Also tested the "decrypt-all" command for a user who had both encrypted and non-encrypted files and it still works fine.

Please review @owncloud/encryption

**Comment by schiesbn** [comment]
looks good now :+1:

**Comment by PVince81** [comment]
:+1: from me as I'm not the original author

**Comment by PVince81** [comment]
Rebased for CI

**Comment by PVince81** [comment]
Ok, forget it, needs more work...

```
PHP Fatal error:  Call to a member function isEncrypted() on array in /ssd/jenkins/workspace/core-ci-linux/database/sqlite/label/SLAVE/lib/private/Encryption/DecryptAll.php on line 217
```

**Comment by PVince81** [comment]
Rebased for CI.

From what I see in the commit list, @schiesbn has fixed the unit test

**Pull request merged**